Cookbook/Proposals/Comments.md

```diff
@@ -0,0 +1,224 @@
+
+In my opinion, our approach should insist on writing always a **self-documented code**. It should be our goal. However, writing self-documented code **doesn't** mean we shouldn't use comments. It means, that commenting our code is a **necessary evil**:
+
+> The proper use of comments is to compensate for our failure to express ourself in code. Note that I used the word failure. I meant it. Comments are always failures. We must have them because we cannot always figure out how to express ourselves without them, but their use is not a cause for celebration.
+    // Robert C. Martin - "Clean Code"
+    
+I think that adding the above section to our style guide would make it clear what's our approach to comments.
+
+### What is a good comment?
+
+To make our StyleGuide more verbose we can also add a list of what kind of comments we think are "good".
+
+**Good comments**:
+
+- **TODO comments**
+    This already has a good definition of our approach in the StyleGuide and I think we are fine with that approach.
+- **Doc headers for public API**
+    Good documentation helps consumers of the SDK to understand what the code does. However, I don't know if it's worth to add a comment to every public function/field, even if a comment is just a repetition. The following comment doesn't add any value (unless it's needed by a doc generation tool if we use any):
+    ```swift
+     /// Supported biometric types
+     public var supportedBiometry: BiometryType
+    ```
+- **Explanation of Intent/Clarification**
+    It's a kind of comment when a developer tries to explain what he's trying to achieve when a code is not verbose enough. Example: Math code, sometimes parsing.
+- **Warning of Consequences**
+    If we had a test which takes an hour to run, it would be nice to have a comment warning about it.
+- **Amplification**
+    Sometimes some piece of code seems to be less important than it really is. In such cases, it's worth to mark it.
+- **Reference to external sources**
+    For example, ISO 639 for country codes.
+
+The above list is what defines a good comment. **However**, we still have to remember that **every** comment **is our failure** to name our code better. If someone gives you a good hint how to remove a comment without sacrificing readability **we should go for it**.
+
+### How can we replace a comment?
+In case of a **what** comments usually it is very easy to replace a comment. Usually, it's all about taking what comment says and create a variable or function with such a name.
+
+
+### How do we comment?
+Even with the proposed approach, we will be forced to comment our code at some point. I would like to propose a way of doing it. The biggest difficulty for me when I read a comment (especially a long one, explaining "why") is to understand to what scope the comment refers to. I think that we should always try to export commented code into a separate method/class and put the comment as a header to this method.
```

Comment:
Even for comments applying to only one line, and especially todo-comments and reference-to-links comments?

For comments regarding a workaround for example I agree. for a comment referencing ISO codes or a jira ticket, not so sure.

Reply:
> For comments regarding a workaround for example I agree.
👍 

>  for a comment referencing ISO codes or a jira ticket, not so sure.
👍 

I agree I wasn't thinking about iSO codes etc. I'll mention that somewhere